src/watch.js

```diff
@@ -0,0 +1,178 @@
+import config from './config.js';
+import datadog from './datadog.js';
+import log from './log.js';
+import ms from 'ms';
+import * as npm from './npm/index.js';
+import saveDocs from './saveDocs.js';
+import * as sentry from './utils/sentry.js';
+
+let loopStart;
+
+/**
+ * Run watch and catchup
+ *
+ *  --- Catchup ?
+ *   If the bootstrap is long or the process has been stopped long enough,
+ *   we are lagging behind few changes.
+ *   catchup() will paginate through changes that we have missed.
+ *
+ *  --- Watch ?
+ *   Watch is "Long Polled. This mode is not paginated and the event system in CouchDB send
+ *     events as they arrive, which is super cool and reactive.
+ *   One gotcha those events arrive at the same rate wether you are watching the last seq or not.
+ *
+ *   Example:
+ *    listener A - up to date
+ *    listener B - few sequences behind
+ *
+ *    package C is updated
+ *
+ *    listener A receive update C
+ *    listener B receive update N
+ *
+ *    listener A is up to date again
+ *    listener B is still few sequences behind and will not receive any other event
+ *      until an other package is updated.
+ *      It will never be up to date because he receive event at the same pace
+ *      as they arrive in listener A, even if it's not the same package
+ *
+ *
+ *  --- We could use catchup with a timeout between poll then?
+ *   yes !
+ *   When we are catched up, we could await between poll and we will receive N changes.
+ *   But long-polling is more efficient in term of bandwidth and more reactive.
+ *
+ */
+async function run(stateManager, mainIndex) {
+  await stateManager.save({
+    stage: 'watch',
+  });
+
+  log.info('🚀  Replicate is up to date, synchronous mode activated');
+
+  await catchup(stateManager, mainIndex);
+  await watch(stateManager, mainIndex);
+}
+
+/**
+ * Loop through all changes that may have been missed
+ *
+ * @param {object} stateManager The state manager
+ * @param {object} mainIndex Algolia index manager
+ */
+async function catchup(stateManager, mainIndex) {
+  let hasCatchedUp = false;
+  while (!hasCatchedUp) {
+    loopStart = Date.now();
+
+    try {
+      const { seq: totalSequence } = await npm.getInfo();
+      const { seq } = await stateManager.get();
+      log.info(
+        '🚀  Replicate: Asking for %d changes since sequence %d',
+        config.replicateConcurrency,
+        seq
+      );
+
+      // Get one chunk of changes from registry
+      const changes = await npm.getChanges({
+        since: seq,
+        limit: config.replicateConcurrency, // eslint-disable-line camelcase
+        include_docs: true, // eslint-disable-line camelcase
+      });
+      hasCatchedUp = await loop(
```

Comment:
Technically `hasCaughtUp`, but this is simpler

Reply:
well, catched is not english so let's go for `hasCaughtUp` :p